src/App.tsx

```diff
@@ -1,24 +1,365 @@
+    completed,
+  }: Todo) {
+    setErrorMessage('');
+
+    const newTamperTodo = {
+      title,
+      userId,
+      completed,
+      id: 0,
+    };
+
+    if (!title.trim()) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => {
+        myInputRef.current?.focus();
+      }, 0);
+
+      return Promise.resolve();
+    }
+
+    setTempTodo(newTamperTodo);
+
+    return todoService.addTodos({
+      title,
+      userId,
+      completed,
+    })
+      .then(
+        (newTodo: Todo) => {
+          if (newTodo.title) {
+            setTodos(currentTodos => [...currentTodos, newTodo]);
+          }
+        },
+      )
+      .catch((err: unknown) => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => {
+          myInputRef.current?.focus();
+        }, 0);
+        throw err;
+      })
+      .finally(() => {
+        setInterval(() => setErrorMessage(''), 3000);
+        setTempTodo(null);
+      });
+  }
+
+  async function deleteTodo(id: number) {
+    setIsLoad(true);
+    const newTamperTodo = todos.find(todo => todo.id === id);
+
+    if (newTamperTodo) {
+      setSelectedTodo(newTamperTodo);
+    }
+
+    return todoService.deleteTodos(id)
+      .then(() => {
+        setTodos(currentPosts => currentPosts.filter(todo => todo.id !== id));
+        setTimeout(() => {
+          myInputRef.current?.focus();
+        }, 0);
+      })
+      .catch((err) => {
+        setTodos(todos);
+        setErrorMessage('Unable to delete a todo');
+
+        throw err;
+      })
+      .finally(() => {
+        setIsLoad(false);
+        setInterval(() => setErrorMessage(''), 3000);
+      });
+  }
+
+  async function updateTitleTodo(updateTodo: Todo) {
+    setSelectedTodo(updateTodo);
+    const isSameTitle = todos.some(todo => todo.title === updateTodo.title);
+
+    if (!updateTodo.title.trim()) {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        await deleteTodo(updateTodo.id);
+      } catch (error) {
+        throw error;
+      }
+
+      return;
+    }
+
+    if (isSameTitle) {
+      return;
+    }
+
+    setErrorMessage('');
+    setIsLoad(true);
+
+    await todoService.updateTodos(updateTodo)
+      .then((newTodo) => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = currentTodos.findIndex(todo => todo.id === updateTodo.id);
+
+          newTodos.splice(index, 1, newTodo);
+
+          return newTodos;
+        });
+        setSelectedTodo(null);
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to update a todo');
+
+        throw error;
+      })
+      .finally(() => {
+        setInterval(() => setErrorMessage(''), 3000);
+        setIsLoad(false);
+      });
+  }
+
+  const handleDubleClick = async (newValue: Todo) => {
+    // eslint-disable-next-line no-useless-catch
+    try {
+      await updateTitleTodo(newValue);
+    } catch (error) {
+      throw error;
```

Comment:
```suggestion
      setErrorMessage('unable to update todo')
```

Reply:
same here

Reply:
There is no sense in try/catch if you throw the same error in catch block

Reply:
<img width="603" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/70220667/5f1b69e1-310f-4e07-93d0-64ddd2311721">